integrations/clickup/client.py

```diff
@@ -0,0 +1,231 @@
+import asyncio
+import time
+from typing import Any, AsyncGenerator, Optional, List, Dict
+from httpx import HTTPStatusError, Timeout
+from loguru import logger
+from port_ocean.utils import http_async_client
+from port_ocean.utils.cache import cache_iterator_result
+
+TEAM_OBJECT = "__team"
+WEBHOOK_EVENTS = [
+    "taskCreated",
+    "taskUpdated",
+    "taskDeleted",
+    "listCreated",
+    "listUpdated",
+    "listDeleted",
+]
+
+MINIMUM_LIMIT_REMAINING = 20  # This limit minus the Semaphore count will be the lower threshold for rate limit in the header
+DEFAULT_SLEEP_TIME = 30  # The Default wait time when limit is hit
```

Comment:
this is quite a lot, can you explain why?

Reply:
I have modified it to five seconds and it is only used as a fail safe if header cannot be found

Reply:
The minimum limit is now dynamically calculated based on the limit allowed for that account by clickup.